src/sysutil.rs

```diff
@@ -191,12 +193,65 @@ impl SysUtil {
                     panic!("Session file '{}' is not UTF-8: {}", path.display(), err)
                 });
 
+                let stem = if let Some(stem) = path.file_stem() {
+                    // Get the stem of the filename of this desktop file.
+                    if let Some(stem) = stem.to_str() {
```

Comment:
Linux supports non-UTF-8 file names and stems. Is there a way to avoid parsing `stem` here as UTF-8? Maybe try using `stems` as `HashSet<OsString>`.

Reply:
> Linux supports non-UTF-8 file names and stems. Is there a way to avoid parsing `stem` here as UTF-8? Maybe try using `stems` as `HashSet<OsString>`.

I could do that, however I lifted this from your code path, which was determining `stem` when `name` parsing failed.

Reply:
Right, but that code is only run in exceptional cases, whereas this one runs for every session. Further, that requirement is because I'm using a `String` for the session name, and fixing that would mean changes everywhere (probably not that hard, but I'm lazy 😛).